Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -109,18 +109,23 @@ RUN curl -sOL https://efa-installer.amazonaws.com/aws-efa-installer-${EFA_INSTAL
     && cd .. \
     && rm -rf aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz aws-efa-installer
 
-# Bootstrap spack compiler installation
-RUN mkdir -p /bootstrap && \
-    cd /bootstrap && \
-    git clone https://github.com/spack/spack spack \
-    && export SPACK_ROOT=/bootstrap/spack \
-    && . spack/share/spack/setup-env.sh \
-    && curl -sOL https://raw.githubusercontent.com/spack/spack-configs/main/AWS/parallelcluster/postinstall.sh \
-    && /bin/bash postinstall.sh -fg -nointel \
-    && spack clean -a \
-    && cd /bootstrap/spack \
-    && find . -type f -maxdepth 1 -delete \
-    && rm -rf bin lib share var /root/.spack
+# Bootstrap spack compiler installation into the eventual installation tree
+# Defined in spack/share/spack/gitlab/cloud_pipelines/configs/config.yaml
+ARG SPACK_ROOT="/bootstrap-compilers/spack"
+#  commit after https://github.com/spack/spack/pull/34821
+ARG SPACK_COMMIT=f27d012
+ARG TARGETARCH
```

Comment:
According to https://github.com/docker/buildx/issues/574#issuecomment-808601577 we don't need to redefine `TARGETARCH`